platform/posix/transport/utest/openssl_utest.c

```diff
@@ -848,10 +875,30 @@ void test_Openssl_Recv_Network_Error( void )
     TEST_ASSERT_TRUE( bytesReceived < 0 );
 
     /* Test that a non-retryable zero error code is converted to -1 by the API. */
+    SSL_pending_ExpectAnyArgsAndReturn( 1 );
     SSL_read_ExpectAnyArgsAndReturn( 0 );
     SSL_get_error_ExpectAnyArgsAndReturn( SSL_ERROR_ZERO_RETURN );
-    bytesReceived = Openssl_Recv( &networkContext, opensslBuffer, BYTES_TO_RECV );
+    bytesReceived = Openssl_Recv( &networkContext, opensslBuffer, 1U );
     TEST_ASSERT_EQUAL( -1, bytesReceived );
+
+    /* Test that a poll error results in the function under test returning -1. */
+    SSL_pending_ExpectAnyArgsAndReturn( 0 );
+    poll_ExpectAnyArgsAndReturn( -1 );
+    bytesReceived = Openssl_Recv( &networkContext, opensslBuffer, 1U );
+    TEST_ASSERT_EQUAL( -1, bytesReceived );
+}
+
+/**
+ * @brief Test that #Openssl_Recv returns 0 when `poll` returns 0.
+ */
+void test_Openssl_Recv_Poll_No_Events( void )
+{
+    int32_t bytesReceived;
+
+    SSL_pending_ExpectAnyArgsAndReturn( 0 );
+    poll_ExpectAnyArgsAndReturn( 0 );
+    bytesReceived = Openssl_Recv( &networkContext, opensslBuffer, 1U );
```

Comment:
Should this also test the case when more than one byte is requested but both `SSL_pending` and `poll` return zero?

Reply:
Sure, doesn't hurt to have it

Reply:
If more than one byte is requested, neither of those functions are called